### DISCUSSION TITLE
Failure of convex decomposition (the mesh self-intersects)

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#6666] [Q&A]
Hello,

Here is a pentagrammic prism:

![](https://user-images.githubusercontent.com/4466543/166666317-3c75eb91-be86-455e-9d2d-53ceb4f42660.png)

It is not possible to triangulate it. The convex decomposition returns only one convex part, so this is wrong because this prism is not convex. Is it due to the self-intersections? And if yes, is the failure systematic when there is self-intersection?

**Comment by pentacular** [comment]
Without knowing what you're actually doing, it is impossible to say.

**Reply by stla** [reply]
:-( There's not 36 functions in CGAL for the convex decomposition, only one.
